include/ddc/kernels/splines/matrix_banded.hpp

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) The DDC development team, see COPYRIGHT.md file
+//
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+#include <memory>
+
+#include "matrix.hpp"
+
+namespace ddc::detail {
+extern "C" int dgbtrf_(
+        int const* m,
+        int const* n,
+        int const* kl,
+        int const* ku,
+        double* a_b,
+        int const* lda_b,
+        int* ipiv,
+        int* info);
+extern "C" int dgbtrs_(
```

Comment:
Not sure if you are aware but this only works with certain compilers. The mangled name will be different for different compilers. But maybe this is not so critical if the plan is to remove LAPACK almost immediately

Reply:
I did not know that, maybe we may wait for a problem to be reported before trying to solve it ? I mean, Lapack is indeed temporary but kokkos-kernel depends on Yuuichi's work too so I cannot tell for how long. But we are the only users of the splines kernel and I guess we will remain for a long time

Reply:
In that case you may want to look into something like https://www.netlib.org/lapack/lapwrapc/

Reply:
My current target  is `tbsv`, and thus it will take a certain amount of time to work on `dgbtrf` and `dgbtrs`. If I understand correctly, this one is called only once in the initialization. Is it correct @blegouix ?

Reply:
No. `dgbtrf` will only be called once in the initialisation but `dgbtrs` is often called at every time step

Reply:
OK. So, we can keep this for `dgbtrf` but not for `dgbtrs`. Anyway, it will take some time. Developing fully portable Linear algebra function is not that easy.

Reply:
Ok. If there is a chance that the methods are kept long term then we definitely should not merge with the assumed name mangling `dgbtrf_`

Reply:
Makes sense

Reply:
Ah yes anyway the factorization methods are definitive

Reply:
@tpadioleau what do we do for this ?

Reply:
I am fine with LAPACKE, it seems to be provided by the LAPACK package, https://netlib.org/lapack/lapacke.html